packages/react-dom-bindings/src/shared/ReactDOMResourceValidation.js

```diff
@@ -16,44 +16,42 @@ export function validateUnmatchedLinkResourceProps(
   currentProps: ?Props,
 ) {
   if (__DEV__) {
-    if (pendingProps.rel !== 'font' && pendingProps.rel !== 'style') {
-      if (currentProps != null) {
-        const originalResourceName =
-          typeof currentProps.href === 'string'
-            ? `Resource with href "${currentProps.href}"`
-            : 'Resource';
-        const originalRelStatement = getValueDescriptorExpectingEnumForWarning(
-          currentProps.rel,
-        );
-        const pendingRelStatement = getValueDescriptorExpectingEnumForWarning(
-          pendingProps.rel,
-        );
-        const pendingHrefStatement =
-          typeof pendingProps.href === 'string'
-            ? ` and the updated href is "${pendingProps.href}"`
-            : '';
-        console.error(
-          'A <link> previously rendered as a %s but was updated with a rel type that is not' +
-            ' valid for a Resource type. Generally Resources are not expected to ever have updated' +
-            ' props however in some limited circumstances it can be valid when changing the href.' +
-            ' When React encounters props that invalidate the Resource it is the same as not rendering' +
-            ' a Resource at all. valid rel types for Resources are "font" and "style". The previous' +
-            ' rel for this instance was %s. The updated rel is %s%s.',
-          originalResourceName,
-          originalRelStatement,
-          pendingRelStatement,
-          pendingHrefStatement,
-        );
-      } else {
-        const pendingRelStatement = getValueDescriptorExpectingEnumForWarning(
-          pendingProps.rel,
-        );
-        console.error(
-          'A <link> is rendering as a Resource but has an invalid rel property. The rel encountered is %s.' +
-            ' This is a bug in React.',
-          pendingRelStatement,
-        );
-      }
+    if (currentProps != null) {
+      const originalResourceName =
+        typeof currentProps.href === 'string'
+          ? `Resource with href "${currentProps.href}"`
+          : 'Resource';
+      const originalRelStatement = getValueDescriptorExpectingEnumForWarning(
+        currentProps.rel,
+      );
+      const pendingRelStatement = getValueDescriptorExpectingEnumForWarning(
+        pendingProps.rel,
+      );
+      const pendingHrefStatement =
+        typeof pendingProps.href === 'string'
+          ? ` and the updated href is "${pendingProps.href}"`
+          : '';
+      console.error(
+        'A <link> previously rendered as a %s but was updated with a rel type that is not' +
+          ' valid for a Resource type. Generally Resources are not expected to ever have updated' +
+          ' props however in some limited circumstances it can be valid when changing the href.' +
+          ' When React encounters props that invalidate the Resource it is the same as not rendering' +
+          ' a Resource at all. valid rel types for Resources are "stylesheet" and "preload". The previous' +
```

Comment:
I fixed an erroneous wording here but otherwise I simply removed the wrapping if block with non-sensical condition test